RecoParticleFlow/PFProducer/plugins/SimPFProducer.cc

```diff
@@ -0,0 +1,295 @@
+// This producer assigns vertex times (with a specified resolution) to tracks.
+// The times are produced as valuemaps associated to tracks, so the track dataformat doesn't
+// need to be modified.
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/View.h"
+
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
```

Comment:
there is RecoParticleFlow/PFSimProducer with SIM dependence
Better move this code there